.github/workflows/pr_cluster_workflow.yml

```diff
@@ -0,0 +1,106 @@
+name: 'Automated Cluster Deployment for Pull Requests'
+
+on:
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - 'config.yaml'
+
+env:
+  TF_CLOUD_ORGANIZATION: "infraheads"
+  TF_API_TOKEN: "${{ secrets.TF_API_TOKEN }}"
+  TF_VAR_proxmox_token_id: "${{ secrets.PROXMOX_TOKEN_ID }}"
+  TF_VAR_proxmox_token_secret: "${{ secrets.PROXMOX_TOKEN_SECRET }}"
+  TF_VAR_github_token: "${{ secrets.TOKEN_GITHUB }}"
+  TF_VAR_netris_controller_host: "${{ vars.NETRIS_CONTROLLER_HOST }}"
+  TF_VAR_netris_controller_login: "${{ secrets.NETRIS_CONTROLLER_LOGIN }}"
+  TF_VAR_netris_controller_password: "${{ secrets.NETRIS_CONTROLLER_PASSWORD }}"
+  TF_VAR_argocd_admin_password: "${{ secrets.ARGOCD_ADMIN_PASSWORD }}"
+  TF_VAR_cluster_name: "turnk8s-${{ github.event.number }}"
+
+jobs:
+  checking_yaml_correctness:
+    runs-on: self-hosted
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+      
+      # * Checks if config.yaml is empty or not. All clusters will be destroyed if config.yaml is empty.
+      # * For draft PRs, the cluster name must be in the format "turnk8s-<PR_NUMBER>",
+      #    as it is assumed to be a test cluster for modifying logic and testing validations and features.
+      # * For non-draft PRs, the cluster name must not be started with "turnk8s-" prefix,
```

Comment:
For cluster creation we use normal PRs. In such PRs the added cluster name should not have turnk8s- prefix.